vortex-datafusion/src/persistent/opener.rs

```diff
@@ -22,6 +27,7 @@ pub(crate) struct VortexFileOpener {
     pub projected_arrow_schema: SchemaRef,
     pub batch_size: usize,
     metrics: VortexMetrics,
+    layout_readers: DashMap<Path, Arc<dyn LayoutReader>>,
```

Comment:
Add a comment here explaining _why_ we're caching layout readers in the first place